### PR TITLE
doc: add v3 data model

### PR DESCRIPTION
**Pull request by xiang90**
/cc @joshix @heyitsanthony @gyuho

### CODE REVIEW

File: Documentation/api_v3.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 
 TODO: API doc
 
+## Data Model
+
+etcd is designed to reliably store infrequently updated data and provide reliable watch queries. etcd exposes previous versions of key-value pairs to support inexpensive snapshots and watch history events (“time travel queries”). A persistent, multi-version, concurrency-control data model is a good fit for these use cases.
+
+etcd stores data in a multiversion [persistent][persistent-ds] key-value store. The persistent key-value store preserves the previous version of a key-value pair when its value is superseded with new data. The key-value store is effectively immutable; its operations do not update the structure in-place, but instead always generates a new updated structure. All  past versions of keys are still accessible and watchable after modification. To prevent the data store from growing indefinitely over time from maintaining old versions, the store may be compacted to shed the oldest versions of superseded data.
```
Comment (gyuho) [inline Documentation/api_v3.md]: `All  past..`  -> `All past...` ?

File: Documentation/api_v3.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+
+### Logical View
+
+The store’s logical view is a flat binary key space. The key space has a byte-sorted index on key so range queries are inexpensive.
```
Comment (heyitsanthony) [inline Documentation/api_v3.md]: `has a lexically sorted index on byte string keys`

**Comment by xiang90** [comment]
@heyitsanthony @gyuho PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by philips** [comment]
@xiang90 I would argue that "infrequently updated" is nebulous and doesn't really add to the explanation. If we qualify infrequent, then maybe. But, right now it isn't really helpful.

**Pull request merged**